src/hooks/useComponentType.ts

```diff
@@ -0,0 +1,10 @@
+import { useMediaQuery, useTheme } from '@material-ui/core';
+
+export const useComponentType = () => {
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('xs'));
+  const isTablet = useMediaQuery(theme.breakpoints.down('md'));
+  const isDesktop = !isMobile && !isTablet;
+  const componentType: 'mobile' | 'tablet' | 'desktop' = isMobile ? 'mobile' : isTablet ? 'tablet' : 'desktop';
+  return { componentType, isMobile, isTablet, isDesktop };
```

Comment:
Do we need to return `isMobile`, `isTablet`, and `isDasktop` if we can use `componentType` to determine the size

Reply:
I think that it is more elegant to return them